h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -203,13 +241,15 @@ class SearchBarController extends Controller {
         this.setState({open: false});
       }
       closeDropdown();
+      this._input.removeEventListener('keydown', setupListenerKeys,
```

Comment:
It does remove the listener. Your JSFiddle doesn't work because you're not passing the same function to `removeEventListener`.

https://jsfiddle.net/swmhkjtm/1/